config/mapprovider.go

```diff
@@ -81,20 +81,39 @@ type ChangeEvent struct {
 
 // Retrieved holds the result of a call to the Retrieve method of a Provider object.
 type Retrieved struct {
+	// Deprecated: Use NewRetrievedFromMap to initialize, and Retrieved.MergeTo to access.
 	*Map
-
-	// CloseFunc specifies a function to be invoked when the configuration for which it was
-	// used to retrieve values is no longer in use and should close and release any watchers
-	// that it may have created.
-	//
-	// If nil, then nothing to be closed.
+	// Deprecated: Use NewRetrievedFromMap to initialize, and Retrieved.Close to access.
 	CloseFunc
 }
 
-// CloseFunc a function to close and release any watchers that it may have created.
+// NewRetrievedFromMap returns a new Retrieved instance that contains a Map data.
+// * cfgMap the Map that will be merged to the given map in the MergeTo.
+// * CloseFunc specifies a function to be invoked when the configuration for which it was
+//   used to retrieve values is no longer in use and should close and release any watchers
+//	 that it may have created.
+func NewRetrievedFromMap(cfgMap *Map, close CloseFunc) Retrieved {
+	return Retrieved{Map: cfgMap, CloseFunc: close}
+}
+
+// MergeTo merges the retrieved configuration into the given Map.
+// Note that the retrieved configuration must not be modified.
+func (r Retrieved) MergeTo(dest *Map) error {
```

Comment:
Sanity check: suppose we have

```
exporters:
   otlp:
     endpoint: ${provider:uri}
```

How would this be supported? IIUC the intention is to have the caller do something like:

```go
m := SomeMap()
r := SomeRetrieved()
subsection, _ := m.Sub("exporters::otlp") // Parent of section?
r.MergeTo(subsection)
```

Is that right? I am not sure I see how this work now that the path has been removed from the interface

Reply:
What I am proposing is to have in the end both "MergeTo" and "MergeToAt". This is to limit the use of the "special case" of empty string path.

Reply:
Ah, I see! Okay, then let's go with this first, yes :)